blockstore/src/main/java/brooklyn/location/blockstore/azure/arm/AzureArmVolumeManager.java

```diff
@@ -0,0 +1,411 @@
+package brooklyn.location.blockstore.azure.arm;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.brooklyn.location.jclouds.JcloudsLocation;
+import org.apache.brooklyn.location.jclouds.JcloudsMachineLocation;
+import org.apache.brooklyn.util.repeat.Repeater;
+import org.apache.brooklyn.util.text.Identifiers;
+import org.apache.brooklyn.util.text.StringShortener;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.brooklyn.util.time.Duration;
+import org.jclouds.azurecompute.arm.AzureComputeApi;
+import org.jclouds.azurecompute.arm.domain.DataDisk;
+import org.jclouds.azurecompute.arm.domain.ResourceGroup;
+import org.jclouds.azurecompute.arm.domain.StorageProfile;
+import org.jclouds.azurecompute.arm.domain.StorageService;
+import org.jclouds.azurecompute.arm.domain.VHD;
+import org.jclouds.azurecompute.arm.domain.VirtualMachine;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineProperties;
+import org.jclouds.azurecompute.arm.features.JobApi;
+import org.jclouds.azurecompute.arm.features.ResourceGroupApi;
+import org.jclouds.azurecompute.arm.features.StorageAccountApi;
+import org.jclouds.azurecompute.arm.features.VirtualMachineApi;
+import org.jclouds.azurecompute.arm.functions.ParseJobStatus;
+import org.jclouds.azurecompute.arm.functions.ParseJobStatus.JobStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import brooklyn.location.blockstore.AbstractVolumeManager;
+import brooklyn.location.blockstore.BlockDeviceOptions;
+import brooklyn.location.blockstore.FilesystemOptions;
+import brooklyn.location.blockstore.api.AttachedBlockDevice;
+import brooklyn.location.blockstore.api.BlockDevice;
+import brooklyn.location.blockstore.api.MountedBlockDevice;
+
+public class AzureArmVolumeManager extends AbstractVolumeManager {
+    
+    // Azure "unmanaged disks" don't support separate creation and attachment phases.
+    // Therefore we override methods like createAttachAndMountVolume. If/when we switch
+    // to using azure's DiskApi then we can refactor this code again. However, that is
+    // not yet supported in jclouds (v2.0.1).
+    
+    // TODO:
+    //  - What should the "lun" number be? Make this configurable?
+    //  - waitForJobToComplete: make timeouts configurable?
+    //  - OSDeviceName: was "/dev/sdc" during testing; in contrast AWS has "/dev/xvdc".
+    //    Is this dependent on the image? Do we need to make the prefix configurable?
```

Comment:
This is dependent on the cloud.
Correct  other clouds we tested it was /dev/sdX and on AWS it's /dev/xvdX. 
Side note. Currently brooklyn-blockstore requires to pass the volume name since for some clouds new volumes begin from `b` and some from `h`. For that best to compare tests for other clouds.